libs/ledger-live-common/src/families/vechain/js-transaction.ts

```diff
@@ -0,0 +1,135 @@
+import { Account, TransactionCommon } from "@ledgerhq/types-live";
+import BigNumber from "bignumber.js";
+import { $Shape } from "utility-types";
+import { DEFAULT_GAS_COEFFICIENT, HEX_PREFIX, MAINNET_CHAIN_TAG } from "./constants";
+import { Transaction } from "./types";
+import { Transaction as ThorTransaction } from "thor-devkit";
+import { calculateFee, estimateGas, generateNonce, getBlockRef } from "./utils/transaction-utils";
+import { VTHO_ADDRESS } from "./contracts/constants";
+import VIP180 from "./contracts/abis/VIP180";
+import { calculateTransactionInfo } from "./utils/calculateTransactionInfo";
+import { isValid } from "./utils/address-utils";
+
+/**
+ * Create an empty VET or VTHO transaction
+ *
+ * @returns {Transaction}
+ */
+export const createTransaction = (): Transaction => {
+  const chainTag = MAINNET_CHAIN_TAG;
+
+  return {
+    family: "vechain",
+    body: {
+      chainTag,
+      // placeholder, if "empty" returns error on send modal open
+      blockRef: "0x0000000000000000",
+      expiration: 18,
+      clauses: [],
+      gasPriceCoef: DEFAULT_GAS_COEFFICIENT,
+      gas: "0",
+      dependsOn: null,
+      nonce: generateNonce(),
+    },
+    amount: BigNumber(0),
+    estimatedFees: BigNumber(0),
+    recipient: "",
+    useAllAmount: false,
+    networkInfo: true,
+  };
+};
+
+/**
+ * Apply patch to a transaction
+ *
+ * @param {Transaction} t
+ * @param {TransactionCommon} patch
+ * @returns patched transaction
+ */
+export const updateTransaction = (
+  t: Transaction,
+  patch: $Shape<TransactionCommon>,
+): Transaction => {
+  return { ...t, ...patch };
+};
+
+/**
+ * Prepare transaction before checking status
+ *
+ * @param {Account} a
+ * @param {Transaction} t
+ */
+export const prepareTransaction = async (
+  account: Account,
+  transaction: Transaction,
+): Promise<Transaction> => {
+  const { amount, isTokenAccount } = await calculateTransactionInfo(account, transaction);
```

Comment:
Seems like this too could be moved in the `if` condition below?

Reply:
no because we need to return the amount below

![Screenshot 2023-09-11 at 01 41 24](https://github.com/LedgerHQ/ledger-live/assets/5785838/1c50d56c-bb3d-4ebf-a55d-c002ca48a5b2)